### PR TITLE
🕷 fix the spider when device changes orientation

### LINKED ISSUE #4925: iPad: Spider View is broken
## 🐛 Bug Report
The physics of the spider view are off in landscape mode, we need to amend it a bit.

## Context
Build/Branch: project/ipad
Phone OS: 12.1
Phone model: ?

## Expected behavior
It should work the same as on mobile.

### PR DESCRIPTION
**Pull request by khoi**
## What's this?
🕷 is now aware of device's orientation

### Relationships

Closes #4925

## How is it done?
Axes rotation math

### Side effects
I hope there is none

## Review hints
Build and play with the spider on a real iPad.

## :squid: Permissions
🦑

### CODE REVIEW

File: Toggl.Daneel/Views/SpiderOnARopeView.cs [hunk 1]
```diff
@@ -255,26 +255,62 @@ private CGPath createCurvedPath(CGPoint anchor, CGPoint[] points)
...
+        private Tuple<double, double> convertXYCoordinate(double originX, double originY,
+            UIInterfaceOrientation orientation)
+        {
+            var x = originX;
+            var y = originY;
+            switch (orientation)
+            {
+                case UIInterfaceOrientation.LandscapeLeft:
+                {
+                    var t = x;
```
Comment (heytherewill) [inline Toggl.Daneel/Views/SpiderOnARopeView.cs]: `(x, y) = (y, -x)`

File: Toggl.Daneel/Views/SpiderOnARopeView.cs [hunk 2]
```diff
@@ -255,26 +255,62 @@ private CGPath createCurvedPath(CGPoint anchor, CGPoint[] points)
...
+                    x = y;
+                    y = -t;
+                    break;
+                }
+                case UIInterfaceOrientation.LandscapeRight:
+                {
+                    var t = x;
```
Comment (heytherewill) [inline Toggl.Daneel/Views/SpiderOnARopeView.cs]: `(x, y) = (-y, x)`

File: Toggl.Daneel/Views/SpiderOnARopeView.cs [hunk 3]
```diff
@@ -255,26 +255,62 @@ private CGPath createCurvedPath(CGPoint anchor, CGPoint[] points)
...
+            return new Tuple<double, double>(x, y);
+        }
         private void processAccelerometerData(CMAccelerometerData data, NSError error)
         {
             if (spiderView == null) return;
 
-            var ax = data.Acceleration.X;
-            var ay = data.Acceleration.Y;
-            var angle = -(nfloat)Math.Atan2(ay, ax);
+            var coordinate = convertXYCoordinate(data.Acceleration.X, data.Acceleration.Y,
```
Comment (heytherewill) [inline Toggl.Daneel/Views/SpiderOnARopeView.cs]:
```csharp
var (ax, ay) = convertXYCoordinate(data.Acceleration.X, data.Acceleration.Y,
    UIApplication.SharedApplication.StatusBarOrientation);
```

File: Toggl.Daneel/Views/SpiderOnARopeView.cs [hunk 4]
```diff
@@ -255,26 +255,62 @@ private CGPath createCurvedPath(CGPoint anchor, CGPoint[] points)
...
+
+                var dx = (nfloat)(ax - previousX);
+                var dy = (nfloat)(ay - previousY);
 
                 var direction = new CGVector(dx, dy);
                 var magnitude = (nfloat)Math.Sqrt(dx * dx + dy * dy);
                 if (magnitude > 0.25f)
                 {
-                    applyForce(direction, magnitude);
+                     applyForce(direction, magnitude);
```
Comment (heytherewill) [inline Toggl.Daneel/Views/SpiderOnARopeView.cs]:
```suggestion
                    applyForce(direction, magnitude);
```

File: Toggl.Daneel/Views/SpiderOnARopeView.cs [hunk 5]
```diff
@@ -255,20 +255,48 @@ private CGPath createCurvedPath(CGPoint anchor, CGPoint[] points)
             return path.CGPath;
         }
 
+        private Tuple<double, double> convertXYCoordinate(double originX, double originY,
```
Comment (kornelijepetak) [inline Toggl.Daneel/Views/SpiderOnARopeView.cs]: Why is the tuple here explicit?

```suggestion
        private (double, double) convertXYCoordinate(double originX, double originY,
```

File: Toggl.Daneel/Views/SpiderOnARopeView.cs [hunk 6]
```diff
@@ -255,20 +255,48 @@ private CGPath createCurvedPath(CGPoint anchor, CGPoint[] points)
...
+                    (x, y) = (-y, x);
+                    break;
+                }
+                case UIInterfaceOrientation.PortraitUpsideDown:
+                    x *= -1;
+                    y *= -1;
+                    break;
+            }
+
+            return new Tuple<double, double>(x, y);
```
Comment (kornelijepetak) [inline Toggl.Daneel/Views/SpiderOnARopeView.cs]:
```suggestion
            return (x, y);
```

**Pull request merged**